tests/test_precedence.py

```diff
@@ -61,3 +77,24 @@ def test_overwrite_value_with_namespace():
     assert subject.key1 == 1
     assert subject.key2 == 2
     assert subject.namespace.key1 == 1
+
+
+def test_union_settings():
+    source = {'key1': 42, 'key2': True}
+    silent = Configuration(source, missing=Missing.SILENT)
+    error = Configuration(source, missing=Missing.ERROR)
+    value = Configuration(source, missing=5)
+
+    assert union(source, source)._missing is NotConfigured
+    assert union(source, source, missing=Missing.SILENT)._missing is NotConfigured
+    assert (silent | source)._missing is union(silent, source)._missing is NotConfigured
+    assert union(silent, error, value, missing=Missing.ERROR)._missing is NoDefault
+    assert (error | source)._missing is union(error, source)._missing is NoDefault
+    assert (value | source)._missing == union(value,source)._missing == 5
```

Comment:
This would be easier to read if the `union` call is always on the left.

```suggestion
    assert union(silent, source)._missing is (silent | source)._missing is NotConfigured
    assert union(silent, error, value, missing=Missing.ERROR)._missing is NoDefault
    assert union(error, source)._missing is (error | source)._missing is NoDefault
    assert union(value, source)._missing == (value | source)._missing == 5
```